TrafficSimulator/Assets/Cam/DefaultCamera.cs

```diff
@@ -0,0 +1,151 @@
+        private bool _hasToggledGameObject;
+        
+        private Vector3 _mousePointDirection;
+        private bool _isNearScreenBorder;
+        
+        private bool _isMovingTowardsToggledObject = false;
+        [SerializeField] private float _togglePanSpeed = 0.3f;
+
+        protected override void Awake()
+        {
+            IsDefault = true;
+            base.Awake();
+        }
+    
+        private void Update()
+        {
+            if (_hasToggledGameObject && !_isMovingTowardsToggledObject) FollowTransform.position = _toggledGameObject.transform.position;
+        }
+
+        public override void HandlePointInput(Vector2 pointPosition)
+        {
+            if (!_enableEdgeScrolling) return;
+            
+            Vector2 viewportPosition = Camera.main.ScreenToViewportPoint(pointPosition);
+            float horizontal = 0f;
+            float vertical = 0f;
+            
+            print(viewportPosition);
+            if (viewportPosition.x < _edgeScrollSensitivty) horizontal = -1f;
+            else if (viewportPosition.x > 1 - _edgeScrollSensitivty) horizontal = 1f;
+
+            if (viewportPosition.y < _edgeScrollSensitivty) vertical = -1f;
+            else if (viewportPosition.y > 1 - _edgeScrollSensitivty) vertical = 1f;
+            
+            _isNearScreenBorder = (horizontal != 0f) || (vertical != 0f);
+            _mousePointDirection = new Vector3(horizontal, vertical);
+        }
+
+        public override void HandleClickInput(InputAction.CallbackContext ctx)
+        {
+            var ray = Camera.main.ScreenPointToRay(Mouse.current.position.ReadValue());
+            if (Physics.Raycast(ray, out var hitInfo) && hitInfo.transform.CompareTag("Vehicle"))
+            {
+                SetToggledGameObject(hitInfo.collider.gameObject);
+            }
+            else
+            {
+                if(_hasToggledGameObject) SetToggledGameObjectToNull();
+            }
+        }
+
+        public override void HandleDoubleClickInput(InputAction.CallbackContext ctx)
+        {
+            var ray = Camera.main.ScreenPointToRay(Mouse.current.position.ReadValue());
+            if (Physics.Raycast(ray, out var hitInfo) && hitInfo.transform.gameObject.Equals(_toggledGameObject))
+            {
+                CameraManager.CameraTarget = FollowTransform;
+                CameraManager.ToggleThirdPersonCamera();
+            }
+        }
+
+        public override void Move(Vector3 direction)
+        {
+            // Ignore the input argument if the mouse is near the screen border
+            if (_isNearScreenBorder) direction = _mousePointDirection.normalized;
+
+            if (direction.sqrMagnitude != 0 && _hasToggledGameObject) SetToggledGameObjectToNull();
+            Vector3 translatedDirection = TranslateDirectionToForward(direction.y, direction.x);
+            FollowTransform.position += translatedDirection * (_movementSpeed * Time.deltaTime);
```

Comment:
```suggestion
            FollowTransform.position += translatedDirection * _movementSpeed * Time.deltaTime;
```

Reply:
This is actually a minor optimization, but could be argued unnecessary if it reduces readability.

Compare 4 multiplications when multiplying the scalars first, with 6 multiplication operations when multiplying the vector with each scalar. However, unnecessary here so will change to suggestion given.